cmd/sql/flow.go

```diff
@@ -90,54 +98,130 @@ func getAstroCloudConfig() (astroDeploymentID, astroWorkspaceID string, err erro
-	astroClient := astro.NewAstroClient(httputil.NewHTTPClient())
-
+func promptAstroEnvironmentConfig(astroDeploymentID, astroWorkspaceID string, isCloud bool) (selectedAstroDeploymentID, selectedAstroWorkspaceID string, err error) {
 	if astroDeploymentID == "" && astroWorkspaceID == "" {
-		fmt.Println("\nWhich Astro Cloud workspace should be associated with " + env + "?")
-		workspace, err := astroWorkspace.GetWorkspaceSelection(astroClient, os.Stdout)
-		if err != nil {
-			return "", "", err
-		}
-		deployments, err := astroDeployment.GetDeployments(workspace, astroClient)
-		if err != nil {
-			return "", "", err
+		if isCloud {
+			selectedAstroDeploymentID, selectedAstroWorkspaceID, err = getAstroDeploymentAndWorkspaceId()
+		} else {
+			selectedAstroDeploymentID, selectedAstroWorkspaceID, err = getSoftwareDeploymentAndWorkspaceId()
 		}
-		deployment, err := astroDeployment.SelectDeployment(deployments, "Which Astro Cloud deployment should be associated with "+env+"?")
 		if err != nil {
-			return "", "", err
+			return
 		}
-		selectedAstroDeploymentID = deployment.ID
-		selectedAstroWorkspaceID = deployment.Workspace.ID
 	} else if astroDeploymentID == "" {
-		deployments, err := astroDeployment.GetDeployments(astroWorkspaceID, astroClient)
-		if err != nil {
-			return "", "", err
+		selectedAstroWorkspaceID = astroWorkspaceID
+		if isCloud {
+			selectedAstroDeploymentID, err = getAstroDeploymentIdWithWorkspaceId(selectedAstroWorkspaceID)
+		} else {
+			selectedAstroDeploymentID, err = getSoftwareDeploymentIdWithWorkspaceId(selectedAstroWorkspaceID)
 		}
-		deployment, err := astroDeployment.SelectDeployment(deployments, "Which Astro Cloud deployment should be associated with "+env+"?")
 		if err != nil {
-			return "", "", err
+			return
 		}
-		selectedAstroDeploymentID = deployment.ID
-		selectedAstroWorkspaceID = astroWorkspaceID
 	} else if astroWorkspaceID == "" {
-		fmt.Println("\nWhich Astro Cloud workspace should be associated with " + env + "?")
-		workspace, err := astroWorkspace.GetWorkspaceSelection(astroClient, os.Stdout)
+		selectedAstroDeploymentID = astroDeploymentID
+		if isCloud {
+			selectedAstroWorkspaceID, err = getAstroWorkspaceId()
+		} else {
+			selectedAstroWorkspaceID, err = getSoftwareWorkspaceId()
+		}
 		if err != nil {
-			return "", "", err
+			return
 		}
-		selectedAstroDeploymentID = astroDeploymentID
-		selectedAstroWorkspaceID = workspace
 	} else {
 		selectedAstroDeploymentID = astroDeploymentID
 		selectedAstroWorkspaceID = astroWorkspaceID
 	}
+	return
+}
+
+func getAstroWorkspaceId() (string, error) {
+	fmt.Println("\nWhich Astro workspace should be associated with " + env + "?")
+	astroClient := astro.NewAstroClient(httputil.NewHTTPClient())
+	workspace, err := astroWorkspace.GetWorkspaceSelection(astroClient, os.Stdout)
+	if err != nil {
+		return "", err
+	}
+	return workspace, nil
+}
+
+func getSoftwareWorkspaceId() (string, error) {
```

Comment:
```suggestion
func getAstroSoftwareWorkspaceId() (string, error) {
```

Reply:
@tatiana I think it's better just to remove astro overall (it doesn't add any information)